core/parallel_state_processor.go

```diff
@@ -288,6 +288,10 @@ func (p *ParallelStateProcessor) Process(block *types.Block, statedb *state.Stat
 
 	deps := GetDeps(blockTxDependency)
 
+	if !VerifyDeps(deps) {
```

Comment:
Let's verify the length of `deps` is equal to the length of transactions in this block

Reply:
[Done](https://github.com/maticnetwork/bor/pull/1132/commits/60b5c4143205586278f0047fe1d4784d56254a6b), thanks!